unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/HighlightsSubSection/HighlightsSubSectionMenu/HighlightsSubSectionComponentController.cs

```diff
@@ -78,6 +78,7 @@ public class HighlightsSubSectionComponentController : IHighlightsSubSectionComp
         this.view.OnEventUnsubscribeEventClicked += UnsubscribeToEvent;
         this.view.OnFriendHandlerAdded += View_OnFriendHandlerAdded;
         this.view.OnViewAllEventsClicked += GoToEventsSubSection;
+        DataStore.i.channels.currentJoinChannelModal.OnChange += OnChannelToJoinChanged;
```

Comment:
you should inject the datastore instead of using the singleton in this class

Reply:
Done in https://github.com/decentraland/unity-renderer/pull/2724/commits/94abfca70f28dbd712029cfdeed0fa1119e44b4a